app/_src/policies/introduction.md

```diff
@@ -1,6 +1,650 @@
+        kind: Mesh
+      default: # Configuration that applies to outgoing traffic
+        key: value
+  from:
+    - targetRef:
+        kind: Mesh
+      default: # Configuration that applies to incoming traffic
+        key: value
+```
+{% endpolicy_yaml %}
+
+{% tip %}
+While some policies can have both a `to` and a `from` section, it is strongly advised to create 2 different policies, one for `to` and one for `from`.
+{% endtip %}
+
+Some policies are not directional and will not have `to` and `from`. Some examples of such policies are [`MeshTrace`](/docs/{{ page.version }}/policies/meshtrace) or [`MeshProxyPatch`](/docs/{{ page.version }}/policies/meshproxypatch).
+For example
+
+{% policy_yaml non-directional %}
+```yaml
+type: NonDirectionalPolicy
+name: my-example
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    key: value
+```
+{% endpolicy_yaml %}
+
+All specs have a **top level `targetRef`** which identifies which proxies this policy applies to.
+In particular, it defines which proxies have their Envoy configuration modified.
+
+{% tip %}
+One of the benefits of `targetRef` policies is that the spec is always the same between Kubernetes and Universal.
+
+This means that converting policies between Universal and Kubernetes only means rewriting the metadata.
+{% endtip %}
+
+## Writing a `targetRef`
+
+`targetRef` is a concept borrowed from [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/). 
+Its goal is to select subsets of proxies with maximum flexibility.
+
+It looks like:
+
+```yaml
+targetRef:
+  kind: Mesh | MeshSubset | MeshService | MeshGateway
+  name: "my-name" # For kinds MeshService, and MeshGateway a name has to be defined
+  tags:
+    key: value # For kinds MeshSubset and MeshGateway a list of matching tags can be used
+  proxyTypes: ["Sidecar", "Gateway"] # For kinds Mesh and MeshSubset a list of matching Dataplanes types can be used
+  labels:
+    key: value # In the case of policies that apply to labeled resources you can use these to apply the policy to each resource
+```
+
+Here's an explanation of each kinds and their scope:
+
+- Mesh: applies to all proxies running in the mesh
+- MeshSubset: same as Mesh but filters only proxies who have matching `targetRef.tags`
+- MeshService: all proxies with a tag `kuma.io/service` equal to `targetRef.name`.{% if_version gte:2.9.x %} This can work differently when using [explicit services](#using-policies-with-meshservice-meshmultizoneservice-and-meshexternalservice){% endif_version %}.
+- MeshGateway: targets proxies matched by the named MeshGateway
+    - Note that it's very strongly recommended to target MeshGateway proxies using this
+      kind, as opposed to MeshService/MeshServiceSubset.
+- MeshServiceSubset: same as `MeshService` but further refine to proxies that have matching `targetRef.tags`. ⚠️This is deprecated from version 2.9.x ⚠️.
+
+Consider the example below:
+
+{% policy_yaml accesslog_example %}
+```yaml
+type: MeshAccessLog
```

Comment:
in this example, we are using both to and from in a single policy, which we strongly advise not to do in one of the previous paragraphs. Should we split it to show good example to users?